### PR TITLE
Add `--silent-breakpad` switch, fixes #3535

### LINKED ISSUE #3535: An option to enable/disable your error reporting dumps
I'll keep this short and sweet.

I would like to see an option to enable/disable the error dumps that are created on a crash.

I run a couple of 24/7 servers for OpenRCT2 and have been having trouble recently with my server manager not being able to start the game back up when the game has crashed with an error dump.

Implementing this sooner rather than later would be great :)

On a side note as well, The OpenRCT2 file in my docs has become soo messy with all the error dumps. Maybe make an error file for the dumps to be put into to keep the file a bit cleaner??

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

**Comment by janisozaur** [comment]
Going this way, a crash dump will still be generated and information printed out to `stdout` (see those `printf` calls), but the UI features (messagebox, opening explorer with files selected) *should** be skipped.

\* keep in mind by the time we reach the crash handler, we will have already crashed, so all bets are off.

**Comment by Gymnasiast** [comment]
Should we really show a message dialog when in headless mode though?

**Comment by janisozaur** [comment]
we would, if the data files were missing, so why is this a special case?

**Comment by Gymnasiast** [comment]
I think it should just print errors to stderr, rather than showing message boxes when in headless mode. That applies to every error: this one, if data files are missing, and in other cases.

**Comment by IntelOrca** [comment]
Yes, fair enough for the switch as some people use non-headless mode and have crazy macros on top to automate things but... it should simply not show any GUI in headless mode, assertions, message boxes, windows etc.

**Comment by janisozaur** [comment]
@IntelOrca do you agree a truly headless mode is a separate thing from this PR?

Seems like @ItsMikey420 and @Epxs have tested it and it works?

File: src/platform/crash.cpp [hunk 1]
```diff
@@ -52,7 +52,10 @@ static bool OnCrash(const wchar_t * dumpPath,
     {
         constexpr const char * DumpFailedMessage = "Failed to create the dump. Nothing left to do. Please file an issue with OpenRCT2 on Github and provide latest save.";
         printf("%s\n", DumpFailedMessage);
-        MessageBoxA(NULL, DumpFailedMessage, OPENRCT2_NAME, MB_OK | MB_ICONERROR);
+        if (!gOpenRCT2SilentBreakpad)
```
Comment (IntelOrca) [inline src/platform/crash.cpp]: `&& !gOpenRCT2Headless`

File: src/platform/crash.cpp [hunk 2]
```diff
@@ -77,6 +80,10 @@ static bool OnCrash(const wchar_t * dumpPath,
         SDL_RWclose(rw);
     }
 
+    if (gOpenRCT2SilentBreakpad)
```
Comment (IntelOrca) [inline src/platform/crash.cpp]: `|| gOpenRCT2Headless`

File: src/cmdline/RootCommands.cpp [hunk 3]
```diff
@@ -160,6 +165,7 @@ exitcode_t CommandLine::HandleCommandDefault()
     }
 
     gOpenRCT2Headless = _headless;
+    gOpenRCT2SilentBreakpad = _silentBreakpad;
```
Comment (IntelOrca) [inline src/cmdline/RootCommands.cpp]: `|| _headless`

**Comment by janisozaur** [comment]
updated, with options help text updated accordingly.

File: src/openrct2.c [hunk 4]
```diff
@@ -59,6 +59,7 @@ utf8 gCustomPassword[MAX_PATH] = { 0 };
 bool gOpenRCT2Headless = false;
 
 bool gOpenRCT2ShowChangelog;
+bool gOpenRCT2SilentBreakpad;
```
Comment (IntelOrca) [inline src/openrct2.c]: Isn't it better to define this in crash.cpp if you have defined the extern in crash.h?

Comment (janisozaur) [inline src/openrct2.c]: I think I tried, but I may have mucked things when it comes to `extern "C" {}` stuff and I don't really have VS handy right now to check this, so I have to rely on appveyor. Feel free to improve that.

**Pull request merged**